### DISCUSSION TITLE
Track ANRs

### DISCUSSION THREAD

**Discussion by hot33331** [getsentry/sentry-dotnet#2174] [Q&A] [answered]
Is it possiblie in Xamarin Forms or MAUI to track ANRs with sentry, or is that something that is on the roadmap?

**Comment by mattjohnsonpint** [comment, marked as answer]
Not in Xamarin Forms, but we already do this for MAUI.  Presently it's on by default for Android only.  To turn it on for iOS, add this to your `UseSentry` options:

```csharp
#if __IOS__
		options.iOS.EnableAppHangTracking = true;
#endif
```

We'll be setting it on by default when we address issue #2148.

Also note that while these will track and report ANRs, the reported stack trace will be in native code - not necessarily the C# code that generated the hang.  We would have some work to do to light that up.   I created an issue for that - #2179.

**Comment by Syed-RI** [comment]
I know this is enabled by default, but I have not seens any data in the dash related to ANR or slow app load times. Am I missing something? Seems stuffs described here: https://docs.sentry.io/product/insights/mobile/mobile-vitals/ doesnt apply to MAUI. Am I missing specific package or config? @Flash0ver , @jamescrosswell @Dhrumil-Sentry

**Reply by jamescrosswell** [reply]
ANRs should work (enabled by default):
https://github.com/getsentry/sentry-dotnet/blob/85c56e2d2e72042889162b120ab89e2099141372/src/Sentry/Platforms/Android/NativeOptions.cs#L32

Mobile Vitals hasn't yet been implemented in the .NET SDK though:
- https://github.com/getsentry/sentry-dotnet/issues/3181

**Reply by Syed-RI** [reply]
yeah, but I have not seen any reported yet
<img width="2730" height="873" alt="Screenshot 2025-08-21 at 09 19 31" src="https://github.com/user-attachments/assets/24cc155b-aa5a-4fe8-984e-63d6a54a75f6" />

Our app cant be that good :trollface: besides I think the MAUI docs needs more TLC. You should also invest on showcasing the setup, capabilities and troubleshooting steps in liaison with a developer advocate in popular video platforms. Ive seen a few basic ones like : https://www.youtube.com/watch?v=9-50zH8fqYA&pp=ygULc2VudHJ5IG1hdWk%3, https://www.youtube.com/watch?v=4HMX5aFZMsc&pp=ygULc2VudHJ5IG1hdWk%3D but we can do better in terms ofshowcasing the other capabilities from Sentry.Android SDK that are in MAUI SDK such as Anrs (and others that I dont know about)

**Reply by jamescrosswell** [reply]
The actual implementation of ANR detection is done in the Android SDK (which the .NET SDK wraps when targeting android)... so there's a bit more detail on when/why it would be triggered here:
- https://docs.sentry.io/platforms/android/configuration/app-not-respond/

Noted re the documentation. This could definitely be improved... If you've got any specific suggestions (page/paragraphs etc. that aren't clear) or features that you didn't find documented anywhere, definitely let us know and we'll make an effort to improve these.

**Reply by Syed-RI** [reply]
> The actual implementation of ANR detection is done in the Android SDK (which the .NET SDK wraps when targeting android)

Yes I thought all the Android SDK features are available on the MAUI SDK as MAUI SDK is wrapping. I suppose thats wrong. 

> If you've got any specific suggestions (page/paragraphs etc. that aren't clear)

Just wishing more features and in depth examples in the doc thats all. Thanks
